source/adminguide/api.rst

```diff
@@ -87,4 +87,93 @@ is also supported.) For metadata type, use one of the following:
 
 -  instance-id. The instance name of the VM
 
+User Data and Meta Data via Config Drive
+----------------------------------------
 
+The user-data service on a Shared or L2 Network can be provided through the
+Virtual Router or through an attached iso called the Config drive.
+
+Config drive is an ISO file that is mounted as a cd-rom on a user VM and
+contains the user VM related userdata, metadata (incl. ssh-keys) and
+password files.
+
+Enable config drive
+~~~~~~~~~~~~~~~~~~~
+To use the config drive the network offering must have the “ConfigDrive”
+provider selected for the userdata service.
+
+If the networkoffering uses ConfigDrive for userdata and the template is
+password enabled, the password string for the VM is placed in the
+vm_password.txt file and it is included in the ISO.
+
+ConfigDrive availability
+~~~~~~~~~~~~~~~~~~~~~~~~
+At VM start the config drive ISO is attached on the 2nd cd/dvd drive of the
+user instance, such that any other ISO image (e.g. boot image or vmware tools)
+is mounted on 1st cd/dvd drive. This means existing functionality of
+supporting 1 cd rom drive is still available.
+
+At Password reset or update of user data, Secondary Storage VM will rebuild the
+ConfigDrive ISO image. That is the existing ISO is mounted on a temporary directory,
+password, userdata or ssh-keys are updated and a new ISO is built from the
+updated directory structure.
+
+In case of a password reset, the new password will be picked-up at VM start.
+To access the updated userdata, the user needs to remount the config drive ISO.
+
+When a VM is stopped, the ConfigDrive network element will trigger the
+Secondary Storage VM to remove the ISO from the secondary storage.
+
+Since the ISO is available on secondary storage, there is no need for an extra
+implementation in case of migration.
```

Comment:
@Spaceman1984 can you check the setting "vm.configdrive.primarypool.enabled", and see if it can be updated here. When this setting is enabled, config drive is hosted from primary storage (for KVM hypervisor).

Reply:
I'm not sure if that setting applies to version 4.15 @sureshanaparti.

Reply:
Done